x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/inference/bulk/BulkInferenceExecutionConfig.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.inference.bulk;
+
+public record BulkInferenceExecutionConfig(int workers, int maxOutstandingRequests) {
+    public static final int DEFAULT_WORKERS = 10;
```

Comment:
how did we arrive to this default value? I see that we are using the same `threadPool` for `InferenceRunner` in `TransportEsqlQueryAction`.

Reply:
See my response to Carlos [here](https://github.com/elastic/elasticsearch/pull/127409#discussion_r2124124100)